resolwe/flow/managers/dispatcher.py

```diff
@@ -836,6 +841,35 @@ async def communicate(self, data_id=None, run_sync=False, save_settings=True):
                 )
             )
 
+    def _lock_inputs_local_storage_locations(self, data):
+        """Lock storage locations of local inputs.
+
+        The reasoning behind is to lock the storage location of inputs so they
+        are not removed while processing data object if some of the inputs is
+        deleted.
+        """
+        dependencies = (
+            Data.objects.filter(
+                children_dependency__child=data,
+                children_dependency__kind=DataDependency.KIND_IO,
+            )
+            .exclude(location__isnull=True)
+            .distinct()
+        )
+
+        for parent in dependencies:
+            file_storage = parent.location
+            if file_storage.has_storage_location(STORAGE_LOCAL_CONNECTOR):
+                # Lock the local storage location.
+                local_storage_location = file_storage.storage_locations.get(
+                    connector_name=STORAGE_LOCAL_CONNECTOR
+                )
+                AccessLog.objects.create(
```

Comment:
Why do you only lock local storage locations? I think it would make sense to lock the local one if it exists and the default one otherwise. This would guarantee that the new Data object will have all inputs available.